### ISSUE TITLE
Add validation on email address on user#update 

### ISSUE THREAD

**Issue by F3PiX** [rails-girls-summer-of-code/rgsoc-teams#885]
Extracted from #413
"Github handle and Email address have a star for indicating a mandatory field, but can be saved without an email address. This happened, because I deleted the email. Might happen, if a candidate wants a different email, deletes it and then forgets to add the new one."

This is partial feature, partial bug ;-)
"The reason behind it is that a User record is indeed created w/o an email address at first: the rest gets pulled from the GH API asynchronously. We should improve that, though. A validation context should fix that: the easy win would be to allow a User to created w/o an email address, just not updated."

**Comment by inioluwaa** [comment]
This issue has been fixed already, should be closed

**Comment by carpodaster** [comment]
@F3PiX can you verify?

**Comment by F3PiX** [comment]
@inioluwaa Thanks for reporting! I love problems that solve themselves ;-) 

I am not aware that this is solved, though. 
Can you please explain a bit how you came to the conclusion? For example: did you find the implemented code, or did you try to update your email address on the site, and were not allowed?

**Comment by ArushiSinghal** [comment]
Is this issue solved?

**Comment by F3PiX** [comment]
I don't think so, @ArushiSinghal. You are very welcome to dive into this. As a first step, make sure that you can reproduce the problem. 
Then, the solution to work on would be: 
> A validation context should fix that: the easy win would be to allow a User to created w/o an email address, just not updated.

Thanks!

**Comment by bajajerk** [comment]
@ArushiSinghal are you working on the issue or should I look into this?

**Comment by ramonh** [comment]
Hi @ArushiSinghal! Thank you for your volunteering to fix this.

Not sure if you've already fixed this, but upon looking into it, I found an HTML5 validation, it seems:

![screen shot 2018-05-07 at 11 05 09](https://user-images.githubusercontent.com/656318/39694155-97b49f6a-51e6-11e8-9cf0-a4ed26e15f94.png)

Could this be browser-dependent? If so, do we need a backend validation as well?
